privacyidea/api/lib/prepolicy.py

```diff
@@ -1431,19 +1404,10 @@ def u2ftoken_allowed(request, action):
             "attestation_subject": x509name_to_string(
                 attestation_cert.get_subject())}
 
-        if user_object:
-            token_user = user_object.login
-            token_realm = user_object.realm
-            token_resolver = user_object.resolver
-        else:
-            token_realm = token_resolver = token_user = None
-
         allowed_certs_pols = policy_object.get_action_values(
             U2FACTION.REQ,
             scope=SCOPE.ENROLL,
-            realm=token_realm,
-            user=token_user,
-            resolver=token_resolver,
+            user_object=request.User if request.User else None,
```

Comment:
One thing: It ``request`` for some reason would not contain the attribute ``User`` I think this would also raise an exception?

If ``request.User`` is empty like ``User()`` I think we can also pass it this way.

So, if we are **certain** that ``request`` **always** has an attribute ``User`` I simply would pass this

    user_object=request.User

and handle the empty User object in the policy methods.